api/v1beta1/machine_types.go

```diff
@@ -42,6 +42,9 @@ const (
 	// MachineDeploymentLabelName is the label set on machines if they're controlled by MachineDeployment.
 	MachineDeploymentLabelName = "cluster.x-k8s.io/deployment-name"
 
+	// MachineControlPlaneNameLabel is the label set on machines if they're controlled by a ControlPlane.
+	MachineControlPlaneNameLabel = "cluster.x-k8s.io/control-plane-name"
+
```

Comment:
I'll open a follow-up issue to rename the other existing label constants.
We can discuss on the new issue but I would suggest:

cluster-api/api/v1beta1:
```go
	ClusterLabelName = "cluster.x-k8s.io/cluster-name" # => ClusterNameLabel
	ClusterTopologyMachineDeploymentLabelName = "topology.cluster.x-k8s.io/deployment-name" # => ClusterTopologyMachineDeploymentNameLabel
	ProviderLabelName = "cluster.x-k8s.io/provider" # => ProviderLabel
	MachineControlPlaneLabelName = "cluster.x-k8s.io/control-plane" # => MachineControlPlaneLabel
	MachineSetLabelName = "cluster.x-k8s.io/set-name" # => MachineSetNameLabel
	MachineDeploymentLabelName = "cluster.x-k8s.io/deployment-name" #  => MachineDeploymentNameLabel
```

clusterctl:
```go
	ClusterctlLabelName = "clusterctl.cluster.x-k8s.io" #  => ClusterctlLabel
	ClusterctlCoreLabelName = "clusterctl.cluster.x-k8s.io/core" #  => ClusterctlCoreLabel
	ClusterctlMoveLabelName = "clusterctl.cluster.x-k8s.io/move" #  => ClusterctlMoveLabel
	ClusterctlMoveHierarchyLabelName = "clusterctl.cluster.x-k8s.io/move-hierarchy" #  => ClusterctlMoveHierarchyLabel
```

The idea is essentially, only include a `Name` in the constant name if the label actually contains a name. And in that case include it before `Label` and not use it as a suffix

(There are no occurences of `AnnotationName`)

Reply:
+1 for this cleanup!

Reply:
Created the issue > https://github.com/kubernetes-sigs/cluster-api/issues/7611